subprojects/diagnostics/src/main/java/org/gradle/configuration/Help.java

```diff
@@ -31,12 +33,21 @@
 
 import javax.inject.Inject;
 
+import java.util.stream.Collectors;
+
 import static org.gradle.internal.logging.text.StyledTextOutput.Style.UserInput;
 
 @DisableCachingByDefault(because = "Produces only non-cacheable console output")
 public class Help extends DefaultTask {
     private String taskPath;
 
+    private ListProperty<TaskDetails> tasks = getObjectFactory().listProperty(TaskDetails.class);
```

Comment:
ListProperty is meant for tracking dynamic additions/removals. Consider using TaskDetailsModel (parent) and regular Property (TaskDetailsModel itself would contain a list of TaskDetail).

Look into finalizeValueOnRead() to cause the value to be computed only once (mapping task selection -> task details model). We will need the value not only during execution, but when storing the cache entry.